support/ebpf/beam_tracer.ebpf.c

```diff
@@ -0,0 +1,103 @@
+#include "bpfdefs.h"
+#include "tracemgmt.h"
+#include "types.h"
+
+// The number of frames to unwind per frame-unwinding eBPF program.
+#define FRAMES_PER_PROGRAM   16
+
+bpf_map_def SEC("maps") beam_procs = {
+  .type = BPF_MAP_TYPE_HASH,
+  .key_size = sizeof(pid_t),
+  .value_size = sizeof(BEAMProcInfo),
+  .max_entries = 256,
+};
+
+static EBPF_INLINE ErrorCode unwind_one_beam_frame(PerCPURecord *record, BEAMProcInfo *info, bool top) {
+  UnwindState *state = &record->state;
+  Trace *trace = &record->trace;
+  u64 sp = state->sp, fp = state->fp, pc = state->pc;
+
+  DEBUG_PRINT("beam: pc: %llx, sp: %llx, fp: %llx", pc, sp, fp);
+
+  bpf_probe_read_user(&state->fp, sizeof(u64), (void*)fp);
+  bpf_probe_read_user(&state->pc, sizeof(u64), (void*)(fp+8));
+  bpf_probe_read_user(&state->sp, sizeof(u64), (void*)(fp+16));
```

Comment:
These should do error checking. The read can fail if trying to read unmapped memory. This can happen if `fp` gets corrupt data / out-of-sync from frames.

We also recently added `unwinder_unwind_frame_pointer` which maybe works? Though it seems a bit different.

As minimum you could also merge `bpf_probe_read_user` reads to one to read all registers once since they are adjacent. This will reduce CPU overhead.